frontend/src/common/Classes/Transaction.ts

```diff
@@ -34,6 +39,11 @@ export default abstract class Transaction {
     return this.creationDate;
   }
 
+  public setCreationDate(date: number) {
+    this.creationDate = date;
+    this.save();
+  }

```

Comment:
Creation date must not be modified, that's why it didn't have a setter. Delete this method and any logic that enables changing it.